pkg/provider/azure_loadbalancer.go

```diff
@@ -3126,6 +3131,14 @@ func (az *Cloud) getPublicIPUpdates(
 				toBeDeleted = true
 			}
 
+			if !wantLb && !toBeDeleted {
```

Comment:
Do we consider doing this in `unbindServiceFromPIP`? (line 3600)

Reply:
looks like we also need to check whether service name is equal to the value in tag, like what we do in `unbindServiceFromPIP`

Reply:
I feel we can remove code related to dns tags in `unbindServiceFromPIP` function, or integrate all tags together in `bindServicesToPIP`. Current looks a little bit confusing.

Reply:
Makes sense. Updated

Reply:
lgtm